sha3-asm/build.rs

```diff
@@ -14,11 +14,46 @@ fn main() {
     eprintln!("selected cryptogams script flavor: {flavor:?}");
     run_perlasm(script, flavor.as_deref(), &sha3);
 
+    // MSVC's provided arm assembler does not support -D, only allowing PreDefine to be used.
+    // Unfortunately these are subtly different from -D, making them difficult to use when there
+    // might be symbol conflicts.
+    //
+    // Instead, we will do a find/replace on the assembly here.
+    if target.is_msvc() && target.is_any_arm() {
```

Comment:
Can you move this to before `compile()`, and use the same array to do `.replace` on the string? Order of args shouldn't matter

Reply:
the underscore renames would rename / match previously renamed `KECCAK_ASM_SHA3_absorb`, so I'm not sure `replace`ing with the same array is a good idea. But this can be moved before `compile`

Reply:
figured something out